src/rust/engine/options/src/args.rs

```diff
@@ -143,25 +145,83 @@ impl Args {
     }
 }
 
+pub(crate) struct ArgsTracker {
+    unconsumed_flags: Mutex<HashMap<Scope, HashSet<String>>>,
```

Comment:
In case it's not obvious (although this is a common Rust idiom), `Arc` of (struct containing a) `Mutex` is what allows thread-safe interior mutability, unlike `RefCell`, which is not `Sync`.